src/qtsoap.h

```diff
@@ -369,16 +287,16 @@ class QT_QTSOAP_EXPORT QtSoapStructIterator
     ~QtSoapStructIterator();
 
     QtSoapQName key() const;
-    QtSoapType *data();
-    const QtSoapType *current() const;
+    std::shared_ptr<QtSoapType> data();
+    const std::shared_ptr<QtSoapType> current() const;
 
     void operator ++();
     bool operator !=(const QtSoapStructIterator &j) const;
     bool operator ==(const QtSoapStructIterator &j) const;
 
 private:
-    QList<QtSmartPtr<QtSoapType> >::Iterator it;
-    QList<QtSmartPtr<QtSoapType> >::Iterator itEnd;
+    QList<std::shared_ptr<QtSoapType> >::Iterator it;
+    QList<std::shared_ptr<QtSoapType> >::Iterator itEnd;
```

Comment:
You are fine here: just use `>>` instead of the ugly leftover from ancient times! :smile: